lib/web_ui/lib/src/engine/embedder.dart

```diff
@@ -752,48 +659,9 @@ void applyGlobalCssRulesToSheet(
   }
 }
 
-/// Miscellaneous statistics collecting during a single frame's execution.
-///
-/// This is useful when profiling the app. This class should only be used when
-/// assertions are enabled and therefore is not suitable for collecting any
-/// time measurements. It is mostly useful for counting certain events.
-class DebugDomRendererFrameStatistics {
-  /// The number of times we reused a previously initialized paragraph ruler to
-  /// measure a paragraph of text.
-  int paragraphRulerCacheHits = 0;
-
-  /// The number of times we had to create a new paragraph ruler to measure a
-  /// paragraph of text.
-  int paragraphRulerCacheMisses = 0;
-
-  /// The number of times we used a paragraph ruler to measure a paragraph of
-  /// text.
-  int get totalParagraphRulerAccesses =>
-      paragraphRulerCacheHits + paragraphRulerCacheMisses;
-
-  /// The number of times a paragraph of rich text was laid out this frame.
-  int richTextLayouts = 0;
-
-  /// The number of times a paragraph of plain text was laid out this frame.
-  int plainTextLayouts = 0;
-
-  @override
-  String toString() {
-    return '''
-Frame statistics:
-  Paragraph ruler cache hits: $paragraphRulerCacheHits
-  Paragraph ruler cache misses: $paragraphRulerCacheMisses
-  Paragraph ruler accesses: $totalParagraphRulerAccesses
-  Rich text layouts: $richTextLayouts
-  Plain text layouts: $plainTextLayouts
-'''
-        .trim();
-  }
-}
-
-/// Singleton DOM renderer.
-DomRenderer get domRenderer => ensureDomRendererInitialized();
+/// The embedder singleton.
+FlutterViewEmbedder get flutterViewEmbedder => ensureFlutterViewEmbedderInitialized();
 
-/// Initializes the [DomRenderer], if it's not already initialized.
-DomRenderer ensureDomRendererInitialized() => _domRenderer ??= DomRenderer();
-DomRenderer? _domRenderer;
+/// Initializes the [FlutterViewEmbedder], if it's not already initialized.
+FlutterViewEmbedder ensureFlutterViewEmbedderInitialized() => _flutterViewEmbedder ??= FlutterViewEmbedder();
+FlutterViewEmbedder? _flutterViewEmbedder;
```

Comment:
OT: Is this some pre-`late` wizardry to control where `_flutterViewEmbedder` is exactly created? Should _flutterViewEmbedder be `late ... = FlutterViewEmbedder();`?

Reply:
No. It used to be `final` (static and top-level variables don't need `late`, they are `late` automatically). However, this makes it a getter, and it makes the code look non-idiomatic when you need to do initialization. We used to have code like this in our tests:

```
domRenderer;
```

But it doesn't look like it's doing anything. You just rely on the magic that's happening in the getter. So I changed it to be a function, which looks much better:

```
ensureFlutterViewEmbedderInitialized();
```